pages/docs/coding-conventions/search-inputs.en-US.mdx

```diff
@@ -0,0 +1,93 @@
+# Search inputs
+
+The following are some general guidelines for search inputs:
+
```

Comment:
```suggestion

- Prefer using server-side search functionality over implementing search-filtering on the client side, as the latter requires fetching the entire data set and can be inefficient. However, client-side search might be the right choice in cases where the server-side search functionality is insufficient or the expected result size is small. In either case, be careful with handling server-side pagination, as most REST resources has a search result [size limit](https://openmrs.atlassian.net/wiki/spaces/docs/pages/25469882/REST+Module) per request. Use the [useOpenmrsPagination](https://github.com/openmrs/openmrs-esm-core/blob/0120ab0a384d7a2adcd90aec88dc224802af6ab2/packages/framework/esm-react-utils/src/useOpenmrsPagination.ts), [useOpenmrsFetchAll](https://github.com/openmrs/openmrs-esm-core/blob/f76380b9886f893d8184451e3ac9f27057ee18d7/packages/framework/esm-react-utils/src/useOpenmrsFetchAll.ts) and [useOpenmrsInfinite](https://github.com/openmrs/openmrs-esm-core/blob/2c36c62216a94589bfde7b5ed46cdeee08b308fb/packages/framework/esm-react-utils/src/useOpenmrsInfinite.ts) hooks to handle server-side pagination for different use cases.

```tsx
  // client side search filtering, `data` contains entire data set when `isLoading` is no longer true.
  const { data, totalCount, isLoading } = useOpenmrsFetchAll<Patient>(`${restBaseUrl}/patient`);
  // do something with the data
```

```tsx
  // server-side search filtering. `data` initially set to first page of results, and changes to data of the specified
  // page when `goTo()` function is called.
  // Use this for UI components that have pagination (like <Table>)
  const { data, totalCount, currentPage, goTo } = useOpenmrsPagination<Patient>(`${restBaseUrl}/patient?q=${searchTerm}`);
  // do something with the data
```


```tsx
  // server-side search filtering. `data` initially set to first page of results and grows to first X pages
  // when `loadMore()` is called X-1 times
  // Use this for UIs with infinite scrolling
  const { data, totalCount, hasMore, loadMore, error} = useOpenmrsInfinite<Patient>(`${restBaseUrl}/patient?q=${searchTerm}`);

  const onScrollToBottom = () => {
    if (hasMore && !error) {
      loadMore();
    }
  }
  // do something with the data
```